src/api/app/services/trigger_controller_service/scm_extractor.rb

```diff
@@ -0,0 +1,56 @@
+module TriggerControllerService
+  # NOTE: this class is coupled to GitHub pull requests events and GitLab merge requests events.
+  class ScmExtractor
+    ALLOWED_GITHUB_ACTIONS = ['opened', 'synchronize', 'closed'].freeze
+    ALLOWED_GITLAB_ACTIONS = ['open', 'update', 'close', 'reopen', 'merge'].freeze
+
+    def initialize(scm, event, payload)
+      # TODO: What should we do when the user sends a wwwurlencoded payload? Raise an exception?
+      @payload = payload
+      @scm = scm
+      @event = event
+    end
+
+    def accepted_event_and_action?
```

Comment:
Allowed or accepted. Please decide.

Reply:
Renamed to `allowed_*" to be consistent with the constant names that the methods use: `ALLOWED_GITHUB_ACTIONS` and `ALLOWED_GITLAB_ACTIONS`.